projects/spectator/src/lib/matchers.ts

```diff
@@ -340,18 +340,29 @@ function isHidden(elOrSelector: HTMLElement | string): boolean {
     return true;
   }
 
+  const hiddenWhen = [
+    el => !(el.offsetWidth || el.offsetHeight || el.getClientRects().length),
+    el => el.style.display === 'none',
+    el => el.style.visibility === 'hidden',
+    el => el.type === 'hidden',
+    el => el.hasAttribute('hidden')
+  ];
+
+  const runningInJsDom = navigator.userAgent.includes('Node.js') || navigator.userAgent.includes('jsdom');
+  if (runningInJsDom) {
+    // When running in JSDOM (Jest), offset-properties and client rects are always reported as 0
+    // - hence, let's take a more "naive" approach here. (https://github.com/jsdom/jsdom/issues/135)
+    hiddenWhen.shift();
+  }
+
   while (el) {
     if (el === document) {
       break;
     }
 
-    if (
-      !(el.offsetWidth || el.offsetHeight || el.getClientRects().length) ||
-      el.style.display === 'none' ||
-      el.style.visibility === 'hidden' ||
-      el.type === 'hidden' ||
-      el.hasAttribute('hidden')
-    ) {
+    const or = (previous, current) => current || previous;
+    const elementIsHidden = hiddenWhen.map(rule => rule(el)).reduce(or, false);
```

Comment:
We can use `elementIsHidden = hiddenWhen.some(Boolean)`

Reply:
Agreed!